armi/reactor/blueprints/assemblyBlueprint.py

```diff
@@ -153,17 +192,67 @@ def _constructAssembly(self, cs, blueprint):
 
         return a
 
-    def _createBlock(self, cs, blueprint, bDesign, axialIndex):
+    def _compileAssemblyMaterialInput(self):
+        """
+        Combine the blockwise and componentwise material inputs for the entire
+        assembly into a single dict `assemblyMaterialInput`. The combined dict
+        has the structure:
+        ```
+        {"": {blockwise_mod_name1: blockwise_mod_list1}, ..., {blockwise_mod_nameN: blockwise_mod_listN},
+         comp_name1: {componentwise_mod_name1: componentwise_mod_list1}, ..., {componentwise_mod_nameN: componentwise_mod_listN},
+         .
+         .
+         .
+         comp_nameN: {componentwise_mod_name1: componentwise_mod_list1}, ..., {componentwise_mod_nameN: componentwise_mod_listN}}
+         ```
+
+        For each `mod_list`, there are expected to be as many entries as there are
+        blocks in the assembly. This may be relaxed in the future by allowing a
+        single value to apply to all blocks without needing to write out a whole list.
+
+        To maintain backwards compatibility, we first assumed that there might be
+        blockwise matMods on the base object, then we add in any blockwise matMods
+        that are explicitly denoted using the `blockwise` attribute.
+        """
+        assemblyMaterialInput = {}
+
+        if self.materialModifications != None:
+            assemblyMaterialInput[""] = {}
+            for modName, modList in self.materialModifications.items():
+                assemblyMaterialInput[""][modName] = modList
+
+            if self.materialModifications.blockwise:
+                for modName, modList in self.materialModifications.blockwise.items():
+                    if modName in assemblyMaterialInput[""].keys():
```

Comment:
no need to say `.keys()` here. The `in` statement checks keys already.

Reply:
Fixed.